# Reading and importing the dataframe

importing pandas and reding the data

In [1]:
import pandas as pd
df = pd.read_csv('powerconsumption.csv')

Print the first few rows

In [2]:
df.head()

Datetime  Temperature  Humidity  WindSpeed  GeneralDiffuseFlows  \
0  1/1/2017 0:00        6.559      73.8      0.083                0.051   
1  1/1/2017 0:10        6.414      74.5      0.083                0.070   
2  1/1/2017 0:20        6.313      74.5      0.080                0.062   
3  1/1/2017 0:30        6.121      75.0      0.083                0.091   
4  1/1/2017 0:40        5.921      75.7      0.081                0.048   

   DiffuseFlows  PowerConsumption_Zone1  PowerConsumption_Zone2  \
0         0.119             34055.69620             16128.87538   
1         0.085             29814.68354             19375.07599   
2         0.100             29128.10127             19006.68693   
3         0.096             28228.86076             18361.09422   
4         0.085             27335.69620             17872.34043   

   PowerConsumption_Zone3  
0             20240.96386  
1             20131.08434  
2             19668.43373  
3             18899.27711  
4             18442.40964

Get information about the DataFrame

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52416 entries, 0 to 52415
Data columns (total 9 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Datetime                52416 non-null  object 
 1   Temperature             52416 non-null  float64
 2   Humidity                52416 non-null  float64
 3   WindSpeed               52416 non-null  float64
 4   GeneralDiffuseFlows     52416 non-null  float64
 5   DiffuseFlows            52416 non-null  float64
 6   PowerConsumption_Zone1  52416 non-null  float64
 7   PowerConsumption_Zone2  52416 non-null  float64
 8   PowerConsumption_Zone3  52416 non-null  float64
dtypes: float64(8), object(1)
memory usage: 3.6+ MB


Get summary statistics

In [4]:
df.describe()

Temperature      Humidity     WindSpeed  GeneralDiffuseFlows  \
count  52416.000000  52416.000000  52416.000000         52416.000000   
mean      18.810024     68.259518      1.959489           182.696614   
std        5.815476     15.551177      2.348862           264.400960   
min        3.247000     11.340000      0.050000             0.004000   
25%       14.410000     58.310000      0.078000             0.062000   
50%       18.780000     69.860000      0.086000             5.035500   
75%       22.890000     81.400000      4.915000           319.600000   
max       40.010000     94.800000      6.483000          1163.000000   

       DiffuseFlows  PowerConsumption_Zone1  PowerConsumption_Zone2  \
count  52416.000000            52416.000000            52416.000000   
mean      75.028022            32344.970564            21042.509082   
std      124.210949             7130.562564             5201.465892   
min        0.011000            13895.696200             8560.081466   
25%        0.122000            26310.668692            16980.766032   
50%        4.456000            32265.920340            20823.168405   
75%      101.000000            37309.018185            24713.717520   
max      936.000000            52204.395120            37408.860760   

       PowerConsumption_Zone3  
count            52416.000000  
mean             17835.406218  
std               6622.165099  
min               5935.174070  
25%              13129.326630  
50%              16415.117470  
75%              21624.100420  
max              47598.326360

Check for missing values and handling missing data

In [5]:
df.isnull().sum()

Datetime                  0
Temperature               0
Humidity                  0
WindSpeed                 0
GeneralDiffuseFlows       0
DiffuseFlows              0
PowerConsumption_Zone1    0
PowerConsumption_Zone2    0
PowerConsumption_Zone3    0
dtype: int64

# description de la dataframe 


La consommation d'énergie par habitant au Maroc est de 0,56 tep (environ 42% en dessous de la moyenne de l'Afrique du Nord), dont environ 900 kWh d'électricité (38% en dessous de la moyenne régionale) (2020). La progression de la consommation totale d'énergie a ralenti entre 2010 et 2019 (+3%/an, contre 4,5%/an sur 2000-2010) et a diminué de 7% en 2020 pour atteindre environ 21 Mtep.

La production nationale d'hydrocarbures est faible. Tous les produits pétroliers sont importés depuis la fermeture de la seule raffinerie du pays en 2015 (200 000 barils/jour). Les importations de produits pétroliers ont rapidement augmenté de 2015 à 2019 (+6%/an) et ont diminué de 12% en 2020 en raison de la crise du COVID. Les données de consommation d'électricité ont été collectées à partir du système SCADA (Supervisory Control and Data Acquisition) d'Amendis, qui est un opérateur de service public chargé de la distribution d'eau potable et d'électricité depuis 2002. L'objectif du réseau de distribution électrique est de desservir les consommateurs basse et moyenne tension dans les régions de Tétouan. À cette fin, la livraison et la distribution de l'énergie électrique du point de livraison à l'utilisateur final, le client, sont assurées par Amendis. L'énergie distribuée provient de l'Office National de l'Électricité et de l'Eau Potable. Après transformation de la haute tension (63 kV) en moyenne tension (20 kV), il est possible de transporter et de distribuer l'énergie.

Étant donné que la consommation d'électricité est si cruciale pour le pays, l'idée est d'étudier l'impact sur la consommation d'énergie. La base de données est exhaustive dans sa démonstration de la consommation d'énergie de la ville de Tétouan au Maroc. Le réseau de distribution est alimenté par 3 stations de zone : Quads, Smir et Boussafou.

Datetime : Date et heure, formaté comme MM/DD/YYYY HH:MM
Temperature : Température de l'air en degrés Celsius (°C)
Humidity : Humidité relative en pourcentage (%)
WindSpeed : Vitesse du vent en mètres par seconde (m/s)
GeneralDiffuseFlows : Flot de rayonnement diffus général en kilowatts par mètre carré (kW/m²) ou une unité similaire
DiffuseFlows : Flot de rayonnement diffus spécifique en kilowatts par mètre carré (kW/m²) ou une unité similaire
PowerConsumption_Zone1 : Consommation d'énergie de la Zone 1 en watts (W) ou kilowatts (kW)
PowerConsumption_Zone2 : Consommation d'énergie de la Zone 2 en watts (W) ou kilowatts (kW)
PowerConsumption_Zone3 : Consommation d'énergie de la Zone 3 en watts (W) ou kilowatts (kW)

Cette première ligne de données fournit donc un instantané de la situation météorologique et de la consommation d'énergie dans les différentes zones à un moment donné. L'analyse de l'ensemble de ces données permettra de dégager des tendances et des insights sur les liens entre les conditions météorologiques et la consommation d'énergie dans la ville de Tétouan.